src/components/main/friends/friend/mod.rs

```diff
@@ -92,12 +109,35 @@ pub fn Friend<'a>(cx: Scope<'a, Props>) -> Element<'a> {
                             icon: Shape::XMark,
                             state: ui_kit::button::State::Danger,
                             on_pressed: move |_| {
-                                let mut multipass = cx.props.account.clone();
-                                let did_to_remove = cx.props.friend.clone();
-                                match multipass.remove_friend(&did_to_remove) {
-                                    Ok(_) => {}
-                                    Err(_) => {
-                                        log::debug!("error removing friend");
+                                let local_state = use_atom_ref(&cx, STATE).clone();
+                                let rg = cx.props.messaging.clone();
+                                let current_chat_exist = local_state.read().selected_chat.clone();
+                                match current_chat_exist {
+                                    Some(_) => {
+                                        let current_chat = local_state.read().selected_chat.and_then(|x| local_state.read().active_chats.get(&x).cloned());
+                                        let current_chat_condition = match current_chat {
+                                            Some(c) => c,
+                                            None => return,
```

Comment:
returning here will mean you don't delete the friend. the friend should probably be deleted in any case.

Reply:
Resolved by the latest commit